### ISSUE TITLE
Error when deploying with GitHub Action

### ISSUE THREAD

**Issue by dhersz** [r-lib/pkgdown#2166]
Hi! I've recently stumbled across an error in my pkgdown GitHub Action which I'm not able to reproduce locally.

Basically, when building the website vignettes (as of [this commit](https://github.com/ipeaGIT/accessibility)), the site build fails with this message:

```
 -- Building articles -----------------------------------------------------------
Writing 'articles/index.html'
Reading 'vignettes/accessibility.Rmd'
-- RMarkdown error -------------------------------------------------------------
--------------------------------------------------------------------------------
Error in `$<-.data.frame`(`*tmp*`, "call_text", value = c("pkgdown::deploy_to_branch(new_process = FALSE)",  : 
  replacement has 15 rows, data has 14
Calls: <Anonymous> ... trace_format -> trace_as_tree -> $<- -> $<-.data.frame
```

Please see the [action logs](https://github.com/ipeaGIT/accessibility/runs/7417363317?check_suite_focus=true#step:6:85) for the full context.

However, I can run this vignette locally without any problems. I also tried running `build_site()` and `deploy_to_branch()` locally and both calls succeeded. I'm running it on the same OS that the actions run on (Ubuntu 20.04), with the most recent rmarkdown and pkgdown versions installed and with all of my package's dependencies updated.

[This](https://github.com/ipeaGIT/accessibility/blob/main/.github/workflows/pkgdown.yaml) is how my pkgdown action looks like and [this](https://github.com/ipeaGIT/accessibility/blob/main/vignettes/accessibility.Rmd) is how the offending vignette looks like.

Happy to help with any debugging that may be needed from my side. Cheers!

**Comment by maelle** [comment]
:wave: @dhersz! I can't reproduce the error locally either. :grimacing: 

What I recommend for debugging:
- create a new branch, say "pkgdownprob"
- transform the pkgdown GHA workflow file so that it runs for this branch too

```yaml
on:
  push:
    branches:
      - main
      - pkgdownprob
```

I've tried various things myself
- removing both vignettes fixes the build but removing only the first vignette is not enough.
- removing the bibliography fields does not work (I thought maybe something was wrong with this YAML field).
- installing tinytex did not work either.
- Installing Pandoc version 2.18 as it's what I have locally.

I'm at a loss as to what could be wrong here. :thinking:

**Comment by cderv** [comment]
````
-- RMarkdown error -------------------------------------------------------------
--------------------------------------------------------------------------------
````
This part is written by **pkgdown** but there should be an error written here:
https://github.com/r-lib/pkgdown/blob/e72fc30f333dc4f50f3f9d1f274fc29d67e34974/R/rmarkdown.R#L40-L46

We don't have any information so that is not helpful. 

Maybe you can create a test workflow that would just render the Rmd vignette using **rmarkdown** to see if the rendering of Rmd is the issue. Aim is to find at which layer an error is encounter. 

Live debugging Github action using https://mxschmitt.github.io/action-tmate/ is another option to work in the same environment than GHA.

**Comment by dhersz** [comment]
Thanks @maelle and @cderv, I'll try following your suggestions to do some local debugging and hopefully can get back to you soon.

> This part is written by **pkgdown** but there should be an error written here:
> 
> https://github.com/r-lib/pkgdown/blob/e72fc30f333dc4f50f3f9d1f274fc29d67e34974/R/rmarkdown.R#L40-L46
> 
> We don't have any information so that is not helpful.

Sorry if this is a big misunderstanding from my part, but shouldn't you use `cnd$message` instead of `cnd$stderr`? Apparently error conditions have the elements `message` and `call`, so `gsub()`ing `cnd$stderr` (which is `NULL`) returns `character(0)`, which would explain why the error message is missing:

``` r
tryCatch(
  stop("oi"),
  error = function(cnd) gsub("oi", "this is an error", cnd$stderr)
)
#> character(0)

tryCatch(
  stop("oi"),
  error = function(cnd) gsub("oi", "this is an error", cnd$message)
)
#> [1] "this is an error"

cnd <- tryCatch(
  stop("oi"),
  error = function(cnd) cnd
)
str(cnd)
#> List of 2
#>  $ message: chr "oi"
#>  $ call   : language doTryCatch(return(expr), name, parentenv, handler)
#>  - attr(*, "class")= chr [1:3] "simpleError" "error" "condition"
```

**Comment by dhersz** [comment]
Trying to debug with tmate (very nice tool, btw). I'm getting this error when trying to run `render_rmarkdown()`:

```r
pkgdown:::render_rmarkdown(pkg, input = input, output = output_file, output_format = format, output_options = options, quiet = FALSE)
#> Reading 'vignettes/accessibility.Rmd'
#> Error in dyn.load(file, DLLpath = DLLpath, ...) : 
#>   unable to load shared object '/home/runner/work/_temp/Library/stringi/libs/stringi.so':
#>   libicui18n.so.60: cannot open shared object file: No such file or directory
#> -- RMarkdown error -------------------------------------------------------------
#> error in callr subprocess--------------------------------------------------------------------------------
#> Error in `pkgdown:::render_rmarkdown()`:
#> ! Failed to render RMarkdown
#> Caused by error:
#> ! error in callr subprocess
#> Caused by error:
#> ! unable to load shared object '/home/runner/work/_temp/Library/stringi/libs/stringi.so':
#>   libicui18n.so.60: cannot open shared object file: No such file or directory
```

There seems to be a problem with stringi installation:

```r
packageVersion("stringi")
#> [1] ‘1.7.8’
library(stringi)
#> Error: package or namespace load failed for ‘stringi’ in dyn.load(file, DLLpath = DLLpath, ...):
#>  unable to load shared object '/home/runner/work/_temp/Library/stringi/libs/stringi.so':
#>   libicui18n.so.60: cannot open shared object file: No such file or directory
```

**Comment by dhersz** [comment]
Yup, my mistake indeed... I started having the error on [this commit](https://github.com/ipeaGIT/accessibility/commit/5346b05ccd81f19692145c9181eb9df08e6050bc) and then, when I tried to fix it at [this one](https://github.com/ipeaGIT/accessibility/commit/18a60bfb07e5d6dd32faeacd7c281e1881e44a9a) I changed the OS but forgot to change the RSPM link... Changing it from `bionic` to `focal` fixed the problem.

From what I could understand from all of this is that `rlang::abort()` redirects the error message to the standard error stream when on a non-interactive session (which is why I could see the error running interactively with tmate but not on the actions log). So I probably had a different error before my first attempt to fix the action, but since I couldn't see the error message I just revamped the whole action and introduced a new issue... oops.

The very weird 

```
Error in `$<-.data.frame`(`*tmp*`, "call_text", value = c("pkgdown::deploy_to_branch(new_process = FALSE)",  : 
  replacement has 15 rows, data has 14
Calls: <Anonymous> ... trace_format -> trace_as_tree -> $<- -> $<-.data.frame
```

message seemed to be related to `rlang::abort()`. I was getting variations of this error ("replacement has xx rows, data has yy") when I tried to run `rlang::last_error()` after the rmarkdown errors. When running interactively `rlang::abort()` defaults to displaying the full "error tree" ([as per docs](https://github.com/r-lib/rlang/blob/4794e8ac026c244140070aa697126435dbfc13bb/R/cnd-abort.R#L1290)), which resulted in that error always being thrown. I haven't been able to reproduce this behavior locally, so I can't really tell what causes it.

In summary, I messed up with the dependencies installation and created a mess hehe. Sorry for wasting your time, closing the issue now. If anything, I think it would be interesting to display `abort()`'s error message in the action log in this case, but not sure if there's anything that can be done from `pkgdown` site of things. Cheers.

**Comment by cderv** [comment]
Glad you found the root cause !

**Comment by maelle** [comment]
Yay, very glad to see this fixed! I'll keep this possible error in mind for future GHA errors I encounter.
